arch/c4_radixcore_components.puml

```diff
@@ -5,87 +5,99 @@
+    Component(utxo_virtual_memory, "UTXO Virtualized Memory", "Component: ???")
 
+    Boundary(app_layer, "Application Layer") {
+        Component(app_rules1, "App Rules 1", "Component: Scrypt", "Optimised DSL for Particle specific constraints")
+        Component(app_rules2, "App Rules 2", "Component: Scrypt", "Optimised DSL for Particle specific constraints")
+        Component(cm_manager, "Constraint Machine Manager", "Component: AtomOS", "Compiles *App Rules* into *Contraint Logic Procedures* and manages the life-cycle of Constraint Machines.")
+
+        Boundary(hardware_validation, "Hardware Validation") {
+            Component(cm, "UTXO Hardware", "Component: Constraint Machine", "Hierarchical Finite State Machine (HFSM).")
+            Component(clp1, "Constraint Logic Procedure 1", "Component: Dynamic", "Compiled HFSM guard condition (from App Rule 1).")
+            Component(clp2, "Constraint Logic Procedure 2", "Component: Dynamic", "Compiled HFSM guard condition (from App Rule 2).")
+
+            Rel_(cm, clp1, "evaluates valid UTXO", "-->")
+            Rel_(cm, clp2, "evaluates valid UTXO", "-->")
+        }
+
+        Rel_(app_rules1, cm_manager, "registers with", "-->")
+        Rel_(app_rules2, cm_manager, "registers with", "-->")
+        Rel_(cm_manager, clp1, "compiles???", "-->")
+        Rel_(cm_manager, clp2, "compiles???", "-->")
+    }
 
-Rel_(ws_app1, nginx, "uses", "-->")
-Rel_(nginx, api, "uses", "-->")
-
-
-Rel_(api, atoms, "serialises/deserialises Atoms to/from", "-->")
-Rel_(api, atomos, "queries/submits Atoms to/from", "-->")
-
-Rel_(atomos, cm, "queries/submits Atoms to/from", "-->")
-Rel_(cm, cscrypt1, "evaluates with Atom", "-->")
-Rel_(cm, cscrypt2, "evaluates with Atom", "-->")
-Rel_(cm, cscryptn, "evaluates with Atom", "-->")
-Rel_(cm, tempo_api, "write/read Atoms to/from", "-->")
-Rel_(cm, crypto, "uses", "-->")
-
-Rel_(tempo_api, vc, "write/read Atoms to/from", "-->")
-Rel_(tempo_api, mass, "write/read Atoms to/from", "-->")
-Rel_(vc, atomstore, "write/read Atoms to/from", "-->")
-Rel_(mass, atomstore, "write/read Atoms to/from", "-->")
+    Rel_(utxo_checker, cm, "validates atom with", "-->")
+    Rel_(cm_manager, utxo_virtual_memory, "defines hw config", "-->")
+    Rel_(cm, spinstate_checker, "validate Atom local state", "-->")
+    Rel_(spinstate_checker, utxo_virtual_memory, "pass-through read UTXO state", "-->")
+    Rel_(atomic_state_manager, spinstate_checker, "validates against global state", "-->")
+
+    Component(tempo, "Physical Memory Manager", "Component: Tempo", "Radix proprietary (patent pending) consensus engine.")
+    Component(atomstore, "UTXO Physical Memory", "Component: Atom Store", "Optimises and abstracts DB access to key/value databases")
+    Component(berkley, "Key/Value Store", "Component: Berkley DB Java Edition", "[[https://docs.oracle.com/cd/E17277_02/html/index.html Berkley DB]] stores Atoms and various metadata.")
```

Comment:
berkley => berkeley 
I think we should make it more clear that the DB implementation could be swapped out without too much effort, so the DB stuff is more of an interface than a hard link to BerkeleyDB